### PR TITLE
Initial Linux Aarch64 port (PPU only)

### PR DESCRIPTION
**Pull request by Nekotekina**
No, nothing works.
image

### CODE REVIEW

**Comment by Nekotekina** [comment]
Wow, PPU interpreter works now.

**Comment by Nekotekina** [comment]
image

File: rpcs3/Emu/RSX/VK/vkutils/device.cpp [hunk 1]
```diff
@@ -452,6 +452,18 @@ namespace vk
 			enabled_features.shaderStorageImageWriteWithoutFormat = VK_FALSE;
 		}
 
+		if (!pgpu->features.shaderClipDistance)
+		{
+			rsx_log.error("Your GPU does not support shader clip distance. Graphics will not render correctly.");
+			enabled_features.shaderClipDistance = VK_FALSE;
+		}
+
+		if (!pgpu->features.shaderStorageBufferArrayDynamicIndexing)
```
Comment (AniLeo) [inline rpcs3/Emu/RSX/VK/vkutils/device.cpp:461]: lavapipe has this implemented on mesa-git and will be part of Mesa 22.0 release in February. It's best if you test lavapipe with mesa-git, Mesa's current stable has an issue that crashes most games on lavapipe (https://gitlab.freedesktop.org/mesa/mesa/-/issues/4855)

Comment (kd-11) [reply rpcs3/Emu/RSX/VK/vkutils/device.cpp:461]: I tested on ubuntu 21.10 aarch64 and it does have all this stuff working. For some reason neko's system does not work right, but yea, this is not really a fix. We do invariably use these features unconditionally and there is no reasonable workaround for it as it is core fixed function stuff. I'll re-review when this is all ready for review.

Comment (Nekotekina) [reply rpcs3/Emu/RSX/VK/vkutils/device.cpp:461]: I use Ubuntu 22 container which seems to have outdated Mesa but there is no PPA for this version yet or I couldn't find it.

**Comment by MSuih** [comment]
There's user reports that Java mode setting got flipped with this PR, causing issues like #5289 to reappear when default settings are used.

**Comment by Nekotekina** [comment]
@Wooty-B I used pretty much out-of-the-box Ubuntu 22.04 container to cross compile for arm64. I think separate Vulkan SDK isn't required.

**Comment by Wooty-B** [comment]
@Nekotekina I'm currently running on Debian 11 aarch64, been playing around and I'm getting further but seems like it's failing related to ARM NEON. I'm not sure if I need to dig around and try to exclude NEON support, but here is the make failure in question:

```
[ 85%] Building CXX object rpcs3/Emu/CMakeFiles/rpcs3_emu.dir/Cell/lv2/sys_interrupt.cpp.o
In file included from /home/wooty/Build/rpcs3/rpcs3/util/simd.hpp:21,
                 from /home/wooty/Build/rpcs3/rpcs3/Emu/Cell/PPUInterpreter.cpp:19:
/usr/lib/gcc/aarch64-linux-gnu/11/include/arm_neon.h: In function ‘v128 gv_rmuladds_hds16(const v128&, const v128&, const v128&)’:
/usr/lib/gcc/aarch64-linux-gnu/11/include/arm_neon.h:10656:1: error: inlining failed in call to ‘always_inline’ ‘int16x8_t vqrdmlahq_s16(int16x8_t, int16x8_t, int16x8_t)’: target specific option mismatch
10656 | vqrdmlahq_s16 (int16x8_t __a, int16x8_t __b, int16x8_t __c)
      | ^~~~~~~~~~~~~
```
Goal would be to get this built locally, however if there's any private builds uploaded anywhere I'd really just like to test a couple of things on this hardware. Thanks again for your ARM64 efforts and feedback!

**Comment by kd-11** [comment]
> @Nekotekina I'm currently running on Debian 11 aarch64, been playing around and I'm getting further but seems like it's failing related to ARM NEON. I'm not sure if I need to dig around and try to exclude NEON support, but here is the make failure in question:
> 
> ```
> [ 85%] Building CXX object rpcs3/Emu/CMakeFiles/rpcs3_emu.dir/Cell/lv2/sys_interrupt.cpp.o
> In file included from /home/wooty/Build/rpcs3/rpcs3/util/simd.hpp:21,
>                  from /home/wooty/Build/rpcs3/rpcs3/Emu/Cell/PPUInterpreter.cpp:19:
> /usr/lib/gcc/aarch64-linux-gnu/11/include/arm_neon.h: In function ‘v128 gv_rmuladds_hds16(const v128&, const v128&, const v128&)’:
> /usr/lib/gcc/aarch64-linux-gnu/11/include/arm_neon.h:10656:1: error: inlining failed in call to ‘always_inline’ ‘int16x8_t vqrdmlahq_s16(int16x8_t, int16x8_t, int16x8_t)’: target specific option mismatch
> 10656 | vqrdmlahq_s16 (int16x8_t __a, int16x8_t __b, int16x8_t __c)
>       | ^~~~~~~~~~~~~
> ```
> 
> Goal would be to get this built locally, however if there's any private builds uploaded anywhere I'd really just like to test a couple of things on this hardware. Thanks again for your ARM64 efforts and feedback!

You must set -DUSE_NATIVE_INSTRUCTIONS=OFF in your cmake command. Also, wrong forum for discussion.

**Comment by Wooty-B** [comment]
@kd-11 Setting -DUSE_NATIVE_INSTRUCTIONS=OFF and -DUSE_SYSTEM_FFMPEG fixed my compile errors, thank you

**Pull request merged**